### PR TITLE
Simplify Chisel tool installation

### PR DESCRIPTION
**Pull request by lbussell**
The newest chisel version lets you install with `go install ...`. This required a manual update to the chisel version but I'm happy to wait for that version to flow in naturally with dependency flow updates.

### CODE REVIEW

**Comment by richlander** [comment]
That looks very nice!

File: eng/dockerfile-templates/runtime-deps/Dockerfile.chiseled-ubuntu [hunk 1]
```diff
@@ -21,23 +20,13 @@ RUN apt-get update \
         ca-certificates \
         file
 
+RUN go install {{chiselUrl}}@@{{chiselRef}}
```
Comment (mthalman) [inline eng/dockerfile-templates/runtime-deps/Dockerfile.chiseled-ubuntu:23]: Would the syntax look the same regardless of whether it was a commit SHA or tag? If so, I think manifest.versions.json should be updated to not have the `commit:` syntax at all for the chisel repo.

File: src/runtime-deps/6.0/jammy-chiseled/arm32v7/Dockerfile [hunk 2]
```diff
@@ -5,19 +5,12 @@ RUN apt-get update \
...
-    && git checkout e92d18b733647e77b6829968c91f8a16d25c1f2d
-
-WORKDIR /opt/chisel
+# Install chisel
+RUN go install github.com/canonical/chisel@50b1cb3cc6a3431407fadcaaf5a5c559220e30e1
 
-RUN go generate internal/deb/version.go \
-    && go build ./cmd/chisel \
-    && cp /opt/chisel/chisel /usr/bin/ \
+RUN git clone --no-checkout https://github.com/canonical/rocks-toolbox /opt/rocks-toolbox \
```
Comment (MichaelSimons) [inline src/runtime-deps/6.0/jammy-chiseled/arm32v7/Dockerfile:11]: Since we are just grabbing a single file would it make sense to curl it instead?   This whole RUN instruction could be simplified to a single command.

**Review by MichaelSimons — COMMENTED** [review]
Nice simplification.

File: src/runtime-deps/6.0/jammy-chiseled/arm32v7/Dockerfile (hunk 2 above)
Comment (richlander) [reply src/runtime-deps/6.0/jammy-chiseled/arm32v7/Dockerfile:11]: You know ... `ADD` supports this.

File: eng/dockerfile-templates/runtime-deps/Dockerfile.chiseled-ubuntu (hunk 1 above)
Comment (lbussell) [reply eng/dockerfile-templates/runtime-deps/Dockerfile.chiseled-ubuntu:23]: Yep. I think it supports any kind of ref.

File: src/runtime-deps/8.0/jammy-chiseled/arm32v7/Dockerfile [hunk 3]
```diff
@@ -5,11 +5,11 @@ RUN apt-get update \
...
         file
 
-# Install chisel
-RUN go install github.com/canonical/chisel@50b1cb3cc6a3431407fadcaaf5a5c559220e30e1
+RUN go install github.com/canonical/chisel/cmd/chisel@50b1cb3cc6a3431407fadcaaf5a5c559220e30e1
 
-RUN git clone --branch v1.0.0 --depth 1 https://github.com/canonical/rocks-toolbox /opt/rocks-toolbox \
-    && cp /opt/rocks-toolbox/chisel-wrapper /usr/bin/
+RUN curl -sSL --output /usr/bin/chisel-wrapper \
+    https://raw.githubusercontent.com/canonical/rocks-toolbox/v1.0.0/chisel-wrapper \
```
Comment (MichaelSimons) [inline src/runtime-deps/8.0/jammy-chiseled/arm32v7/Dockerfile:11]: This should be indented or included in the previous line.

This is a bit orthogonal to your change, but it feels like this should be pinned for the same reasons the chisel tool is pinned.

Comment (MichaelSimons) [reply src/runtime-deps/8.0/jammy-chiseled/arm32v7/Dockerfile:11]: Nevermind my pinned comment, I see this is a release reference.

Comment (lbussell) [reply src/runtime-deps/8.0/jammy-chiseled/arm32v7/Dockerfile:11]: Not sure what you mean by pinned. The version/commit of rocks-toolbox is managed by the update-dependencies tool, if that's what you mean.

File: src/runtime-deps/6.0/jammy-chiseled/arm32v7/Dockerfile (hunk 2 above)
Comment (lbussell) [reply src/runtime-deps/6.0/jammy-chiseled/arm32v7/Dockerfile]: To humor @richlander I tried it with `ADD`... it can just as easily be reverted 🙂

**Review by MichaelSimons — APPROVED** [review]

**Comment by lbussell** [comment]
The chisel tool has GitHub releases now, so I'm using that. However I think this PR will fail currently due to https://github.com/canonical/chisel/issues/96.

File: eng/dockerfile-templates/runtime-deps/Dockerfile.chiseled-ubuntu [hunk 4]
```diff
@@ -21,24 +19,11 @@ RUN apt-get update \
         ca-certificates \
         file
 
-RUN {{InsertTemplate("../Dockerfile.git-clone", [
```
[the remark this answers is not in the document]
Comment (mthalman) [reply eng/dockerfile-templates/runtime-deps/Dockerfile.chiseled-ubuntu:53]: This is still relevant.

File: eng/update-dependencies/UpdateDependencies.cs [hunk 5]
```diff
@@ -409,7 +409,7 @@ private static async Task<IEnumerable<IDependencyUpdater>> GetUpdatersAsync()
             // (e.g. sha updater requires the version numbers to be updated within the Dockerfiles)
 
             JObject minGitRelease = await GitHubHelper.GetLatestReleaseAsync("git-for-windows", "git");
-            string chiselRef = await GitHubHelper.GetLatestCommitAsync("canonical", "chisel", "main");
```
Comment (mthalman) [inline eng/update-dependencies/UpdateDependencies.cs:412]: `GetLatestCommitAsync` is no longer reference anymore. That method can be deleted.

**Review by mthalman — APPROVED** [review]
